volto/i18n.rst

```diff
@@ -0,0 +1,242 @@
+.. _i18n-label:
+
+====================
+Internationalization
+====================
+
+Making Text Translatable
+========================
+
+Volto uses :file:`react-intl` to translate strings. We start by importing the
+component:
+
+::
+
+    import { FormattedMessage } from 'react-intl';
+
+Then we can use the :file:`FormattedMessage` component to translate a string:
+
+::
+
+    <FormattedMessage
+      id="Some string"
+      defaultMessage="Some string"
+    />
+
+Extracting i18n Strings
+=======================
+
+Volto provides an i18n extraction script to get all translatable strings from
+your application. Run the following:
+
+::
+
+    $ yarn i18n
+
+This will generate the following output:
+
+::
+
+    Extracting messages from source files...
+    Synchronizing messages to pot file...
+    Synchronizing messages to po files...
+    Generating the json files...
+    done!
+
+As the description suggests it will first extract all messages from the source
+files into a json file. Then it will synchronize the extracted messages with the
+.pot master file and with all the .po files found in the project. After that it
+will generate the final .json files which are used in the application. If you
+want to translate the strings just edit the .po files and run the script again.
+This script will combine the messages located in Volto itself and the current
+project, and combine them into the json files.
+
```

Comment:
One sentence per line, please use paragraphs we want the docs also looking nice and readable :)

Reply:
Additionally, use some markup around file extensions to indicate they are special. For example:

```
``.json``
```